### PR TITLE
Performance optimizations for map and binary compartment reports

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by eile** [comment]
@hernando : 6db9103 should have been part of the release, you might have missed it elsewhere?

File: brion/plugin/compartmentReportMap.cpp [hunk 1]
```diff
@@ -325,52 +344,48 @@ floatsPtr CompartmentReportMap::loadFrame( const float time ) const
...
 
     const std::string& scope = _getScope( _uri );
+    const std::string& index = std::string( "_" ) +
+                               std::to_string( _getFrameNumber( time ));
+
     floatsPtr buffer( new floats( getFrameSize( )));
-    floats::iterator iter = buffer->begin();
-    const size_t index = _getFrameNumber( time );
-    std::map< uint32_t, size_t > sizeMap;
+    float* const ptr = buffer->data();
```
Comment (rdumusc) [inline brion/plugin/compartmentReportMap.cpp]: only used inside the lambda below, can be removed (or at least move down)

File: brion/plugin/compartmentReportMap.cpp [hunk 2]
```diff
@@ -371,16 +391,37 @@ floatsPtr CompartmentReportMap::loadFrame( const float time ) const
...
 
+#ifdef BRION_USE_OPENMP
+    lunchbox::a_ssize_t taken;
+    omp_set_num_threads( _stores.size( ));
+#else
     size_t taken = 0;
-    const auto takeValue = [ ptr, &offsetMap, &taken ]
-        ( const std::string& key, char* data, const size_t size )
+#endif
+#pragma omp parallel
```
Comment (rdumusc) [inline brion/plugin/compartmentReportMap.cpp]: even if not BRION_USE_OPENMP..?

File: brion/plugin/compartmentReportBinary.cpp [hunk 3]
```diff
@@ -223,7 +223,7 @@ floatsPtr CompartmentReportBinary::loadFrame( const float timestamp ) const
     if( _header.byteswap )
     {
 #pragma omp parallel for
-        for( int32_t i = 0; i < _header.numCompartments; ++i )
+        for( ssize_t i = 0; i < ssize_t( _comps ); ++i )
```
Comment (rdumusc) [inline brion/plugin/compartmentReportBinary.cpp]: signed size_t ?

File: brion/plugin/compartmentReportMap.cpp (hunk 1 above)
Comment (eile) [inline brion/plugin/compartmentReportMap.cpp]: done

File: brion/plugin/compartmentReportMap.cpp (hunk 2 above)
Comment (eile) [inline brion/plugin/compartmentReportMap.cpp]: Yes, pragma will be ignored if openmp is not "on"

File: brion/plugin/compartmentReportBinary.cpp (hunk 3 above)
Comment (eile) [inline brion/plugin/compartmentReportBinary.cpp]: openmp insists on signed variables for parallel for.

Comment (eile) [inline brion/plugin/compartmentReportBinary.cpp]: _comps is size_t, so ssize_t is closer than int32_t.

File: brion/plugin/spikeReportFile.cpp [hunk 4]
```diff
@@ -22,6 +22,7 @@
 
 #include <lunchbox/log.h>
 #include <lunchbox/stdExt.h>
+#include <boost/algorithm/string.hpp>
```
Comment (eile) [inline brion/plugin/spikeReportFile.cpp]: neither can I, removed. ;)

Comment (eile) [inline brion/plugin/spikeReportFile.cpp]: good catch

**Pull request merged**